### ISSUE TITLE
etcd volumes should handle SELinux

### ISSUE THREAD

**Issue by joshwget** [rancher/rancher#8071]
Adding `:z` should be fine for `etcd:/pdata` and `/var/etcd/backups:/data-backup` but it looks like Docker doesn't support `:z` on anonymous volumes. The best solution to this is probably to migrate `/data` to be either a Docker volume or bind mount.

**Comment by LLParse** [comment]
The data sidekick exists solely for upgrading from _really old_ template versions - we could just get rid of it and be done with it. The sidekick is causing scheduling problems as well, see https://github.com/rancher/rancher/issues/8070#issuecomment-284516225

**Comment by deniseschannon** [comment]
Template changes for SELinux in k8s template version: `v1.5.4-rancher2` available on the `v1.6.0-dev` branch

**Comment by deniseschannon** [comment]
@sangeethah I believe this was already validated but it wasn't officially assigned to test yet.

**Comment by galal-hussein** [comment]
Rancher server: master build
kubernetes version: rancher/k8s:v1.6.1-rancher1-1

@deniseschannon The kubernets stack started on RHEL 7.3 hosts with SELINUX enforced, however the kubelet containers are showing the following error:
```

4/12/2017 10:40:03 PMI0412 19:40:03.656270    5560 kubelet.go:1752] skipping pod synchronization - [Failed to start ContainerManager mkdir /sys/fs/cgroup/cpuacct,cpu: read-only file system]
4/12/2017 10:40:08 PMI0412 19:40:08.656550    5560 kubelet.go:1752] skipping pod synchronization - [Failed to start ContainerManager mkdir /sys/fs/cgroup/cpuacct,cpu: read-only file system]
4/12/2017 10:40:13 PMI0412 19:40:13.656759    5560 kubelet.go:1752] skipping pod synchronization - [Failed to start ContainerManager mkdir /sys/fs/cgroup/cpuacct,cpu: read-only file system]
4/12/2017 10:40:18 PMI0412 19:40:18.657020    5560 kubelet.go:1752] skipping pod synchronization - [Failed to start ContainerManager mkdir /sys/fs/cgroup/cpuacct,cpu: read-only file system]
4/12/2017 10:40:23 PMI0412 19:40:23.657242    5560 kubelet.go:1752] skipping pod synchronization - [Failed to start ContainerManager mkdir /sys/fs/cgroup/cpuacct,cpu: read-only file system]
```
And the nodes are in NotReady state

**Comment by sangeethah** [comment]
@galal-hussein , can you retest this issue with K8s template - rancher v1.6 available on the v1.6.0-dev branch which has k8s v1.5.4?

**Comment by galal-hussein** [comment]
Rancher server: v1.6.0-rc4
Kubernetes version: v1.5.4-rancher1-4.1

I was able to launch kubernetes stack and run several pods/deployments/services on RHEL setup with SELinux enforced.

I followed the documentation link http://docs.rancher.com/rancher/v1.6/en/installing-rancher/selinux/ to use SELinux with Rancher environment.
